executor/kava/guide.md

```diff
@@ -0,0 +1,265 @@
+
+```bash
+Committed at block 332 (tx hash: C3D31985F37BB892883C91582D9D1235CA802E87DABCBF064BAB91E8D057696F, response: {Code:0 Data:[134 192 154 45 198 112 54 2 217 65 59 208 153 156 212 105 166 72 239 14 19 136 176 188 147 245 107 229 212 214 157 109] Log:Msg 0: swapID: 86c09a2dc6703602d9413bd0999cd469a648ef0e1388b0bc93f56be5d4d69d6d Info: GasWanted:0 GasUsed:0 Events:[{Type: Attributes:[{Key:[115 101 110 100 101 114] Value:[98 110 98 49 117 114 102 101 114 109 99 103 57 50 100 119 113 51 54 53 55 50 99 120 52 120 103 56 52 119 112 107 51 108 102 112 107 115 114 53 103 55] XXX_NoUnkeyedLiteral:{} XXX_unrecognized:[] XXX_sizecache:0} {Key:[114 101 99 105 112 105 101 110 116] Value:[98 110 98 49 119 120 101 112 108 121 119 55 120 56 97 97 104 121 57 51 119 57 54 121 104 119 109 55 120 99 113 51 107 101 52 102 56 103 101 57 51 117] XXX_NoUnkeyedLiteral:{} XXX_unrecognized:[] XXX_sizecache:0} {Key:[97 99 116 105 111 110] Value:[72 84 76 84] XXX_NoUnkeyedLiteral:{} XXX_unrecognized:[] XXX_sizecache:0}] XXX_NoUnkeyedLiteral:{} XXX_unrecognized:[] XXX_sizecache:0}] Codespace: XXX_NoUnkeyedLiteral:{} XXX_unrecognized:[] XXX_sizecache:0})
+```
+
+Query the new swap on bnbchain by its ID
+
+```bash
+tbnbcli token query-swap \
+  --swap-id ${BNBCHAIN_SWAP_ID} \
+  --chain-id Binance-Chain-Nile \
+  --node ${BNB_RPC_URL} \
+  --trust-node
+```
+
+The deputy process will create an atomic swap on kava with the same information after _n_ blocks
+
+```bash
+# The deputy will log the expected other_chain_swap_id
+2020-03-26 15:09:59 INFO sendOtherHTLT send chain KAVA HTLT tx success, other_chain_swap_id=da89ae0c4f341ffa38345c635725bcc0d4e221b807fc7f143fabdd1e13c3b4d5, tx_hash=A3DC82B10373B30D00D30BE253DF34DADF0D57CCEAD319F09DDD3553ED2B36FC
+```
+
+The `other_chain_swap_id` and the generated random number from above can be used to claim the swap on kava.
+
+## kava
+
+Claim atomic swap
+
+```bash
+kvcli tx bep3 claim ${KAVA_SWAP_ID} ${RANDOM_NUMBER} --from user
+# enter password
+```
+
+Funds will be transferred to the intended recipient's kava address. The claim should output a tx result log containing:
+
+```bash
+logs:
+- msgindex: 0
+  success: true
+  log: ""
+  events:
+  - type: message
+    attributes:
+    - key: action
+      value: claimAtomicSwap
+```
+
+After _n_ blocks the deputy will relay the successful claim to bnbchain, closing the swap:
+
+```bash
+# the deputy will log an INFO message similar to:
+2020-03-26 15:17:17 INFO sendBEP2Claim send bep2 claim tx success, bnb_swap_id=4c4abc3fcc7a7e9b4f7d586f439c16312a20f49ab8129a00a86dd54257d79b6f, random_number=0x90e2cbb1a04a24553736adcee3c7862b536e7afaa34634d65f397812702030f2, tx_hash=37A4AB493F607041CE1367BC3FBDE382629D60808A07203C44A74A21D9D2D19A
+```
+
+# Transfer bnb from kava to bnbchain
+
+## kava
+
+Create a new Atomic Swap
+
+```bash
+kvcli tx bep3 create ${KAVA_DEPUTY_ADDRESS} ${BNBCHAIN_USER_ADDRESS} ${BNBCHAIN_DEPUTY_ADDRESS} now 1111111bnb 1111111bnb 360 true --from user
+```
```

Comment:
I'm confused about this command. On bnbchain command, there are two address: recipient-addr and recipient-other-chain. However, there are three addresses on the above comand? What is the purpose of `BNBCHAIN_DEPUTY_ADDRESS`?